openwrt-openwisp-monitoring/files/lib/openwisp/interfaces.lua

```diff
@@ -0,0 +1,199 @@
+-- retrieve interfaces information
+local utils = require('openwisp.monitoring_utils')
+
+local cjson = require('cjson')
+local nixio = require('nixio')
+local nixio_data = nixio.getifaddrs()
+
+local uci = require('uci')
+local uci_cursor = uci.cursor()
+
+local ubus_lib = require('ubus')
+local ubus = ubus_lib.connect()
+if not ubus then
+    error('Failed to connect to ubusd')
+end
+local interface_data = ubus:call('network.interface', 'dump', {})
+
+local interfaces = {}
+
+local specialized_interfaces = {
+    modemmanager = function(_, interface)
+        local modem = uci_cursor.get('network', interface['interface'], 'device')
+        local info = {}
+
+        local general = io.popen('mmcli --output-json -m '..modem):read("*a")
```

Comment:
@nemesisdesign Why do we have to force usind modem manger here? Wouldn't it be better if we were more flexible here? On our system we use the openwrt proto 3g, uqmi or mbim and not the modem-manger.

Reply:
@feckert it would be great to support other ways other than modem.manager to get this information!
I suggest to discuss this in a separate issue since it's not something that Kapil can do because it is out of scope for this project.
Could you open a new issue and give some info on how to retrieve signal data with the packages you're currently using and what kind of output is returned?

If this is something your company would be interesting in using, we could invest some time in it.
Right now we are able to visualize this information in the device status tab, and in https://github.com/openwisp/openwisp-monitoring/pull/294 we're working on generating a few charts as well.

Reply:
@nemesisdesign done https://github.com/openwisp/openwrt-openwisp-monitoring/issues/36